src/inference/tests/unit/cpu_map_parser.cpp

```diff
@@ -1299,6 +1299,418 @@ INSTANTIATE_TEST_SUITE_P(CPUMap,
                                          freq_1sockets_8cores_1_hyperthreading,
                                          freq_1sockets_4cores));
 
+struct LinuxNumactlTestCase {
+    std::vector<int> phy_core_list;
+    std::vector<std::vector<int>> input_proc_type_table;
+    std::vector<std::vector<int>> input_cpu_mapping_table;
+    int _sockets;
+    int _cores;
+    std::vector<std::vector<int>> _proc_type_table;
+    std::vector<std::vector<int>> _cpu_mapping_table;
+};
+
+class LinuxCpuMapNumactrlTests : public CommonTestUtils::TestsCommon,
+                                 public testing::WithParamInterface<std::tuple<LinuxNumactlTestCase>> {
+public:
+    void SetUp() override {
+        const auto& test_data = std::get<0>(GetParam());
+
+        int test_sockets = 0;
+        int test_cores = 0;
+        auto test_proc_type_table = test_data.input_proc_type_table;
+        auto test_cpu_mapping_table = test_data.input_cpu_mapping_table;
+
+        update_valid_processor_linux(test_data.phy_core_list,
+                                     test_sockets,
+                                     test_cores,
+                                     test_proc_type_table,
+                                     test_cpu_mapping_table);
+
+        ASSERT_EQ(test_data._sockets, test_sockets);
+        ASSERT_EQ(test_data._cores, test_cores);
+        ASSERT_EQ(test_data._proc_type_table, test_proc_type_table);
+        ASSERT_EQ(test_data._cpu_mapping_table, test_cpu_mapping_table);
+    }
+};
+
+LinuxNumactlTestCase numactrl_2sockets_20cores_hyperthreading_1 = {
```

Comment:
general ask for such kind of tests: lets add inlined comments that explain each parameter in that structure (for the first set only will be enough). It will dramatically simplify understanding of the tests.

Reply:
updated